os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -247,12 +256,122 @@ static int assert_tracecallback(FAR struct usbtrace_s *trace, FAR void *arg)
 }
 #endif
 
+/****************************************************************************
+ * Name: check_assert_location
+ ****************************************************************************/
+static void check_assert_location(uint32_t *sp, bool *is_irq_assert)
+{
+	if (g_irq_num >= 0) {
+		/* Assert in irq */
+		*is_irq_assert = true;
+		lldbg("Code asserted in IRQ state!\n");
+		lldbg("IRQ num: %d\n", g_irq_num);
+		lldbg("IRQ handler: %08x \n", g_irqvector[g_irq_num].handler);
+#ifdef CONFIG_DEBUG_IRQ_INFO
+		lldbg("IRQ name: %s \n", g_irqvector[g_irq_num].irq_name);
+#endif
+	} else {
+		/* Assert in user thread */
+		lldbg("Code asserted in normal thread!\n");
+		if (CURRENT_REGS) {
+			/* If assert is in user thread, but current_regs is not NULL,
+			 * it means that assert happened due to a fault. So, we want to
+			 * reset the sp to the value just before the fault happened
+			 */
+			*sp = CURRENT_REGS[REG_R13];
+		}
+	}
+}
+/****************************************************************************
+ * Name: check_sp_corruption
+ ****************************************************************************/
+#if CONFIG_ARCH_INTERRUPTSTACK > 7
+static void check_sp_corruption(uint32_t sp, uint32_t *stackbase, uint32_t *stacksize, uint32_t istackbase, uint32_t istacksize, bool is_irq_assert, bool *is_sp_corrupt)
+#else
+static void check_sp_corruption(uint32_t sp, uint32_t *stackbase, uint32_t *stacksize, bool is_irq_assert, bool *is_sp_corrupt)
+#endif
```

Comment:
How about using first one only regardless of the CONFIG_ARCH_INTERRUPTSTACK?
I think we can do this like log printing.
```
uint32_t istacksize;
#if CONFIG > 7
istacksize = xxxx;
#else
istacksize = 0xFFFFFFFF; // 0xFFFFFFFF means no irq stack
#endif

check_sp_corruption(......, istacksize, ....);
```

Reply:
Looks good to me. I will implement it.